minio/helpers.py

```diff
@@ -351,16 +357,33 @@ def is_valid_bucket_name(bucket_name):
     :return: True if the bucket is valid. Raise :exc:`InvalidBucketError`
        otherwise.
     """
+    # Verify bucket name is not empty
+    bucket_name = str(bucket_name).strip()
+    if bucket_name == '':
+        raise InvalidBucketError('Bucket name cannot be empty.')
+
     # Verify bucket name length.
     if len(bucket_name) < 3:
         raise InvalidBucketError('Bucket name cannot be less than'
                                  ' 3 characters.')
     if len(bucket_name) > 63:
-        raise InvalidBucketError('Bucket name cannot be more than'
+        raise InvalidBucketError('Bucket name cannot be greater than'
                                  ' 63 characters.')
-    if '..' in bucket_name:
-        raise InvalidBucketError('Bucket name cannot have successive'
-                                 ' periods.')
+
+    match = _VALID_IP_ADDRESS.match(bucket_name)
+    if match:
+        raise InvalidBucketError('Bucket name cannot be an ip address')
+
+    unallowed_successive_chars = ['..', '.-', '-.']
+    if any(x in bucket_name for x in unallowed_successive_chars):
+        raise InvalidBucketError('Bucket name contains invalid '
+                                 'successive chars ("..", ".-", "-.").')
```

Comment:
More easier to do this:
`'success chars ' + str(unallowed_successive_chars)`

Reply:
Done.